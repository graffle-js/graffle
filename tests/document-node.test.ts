import { gql } from 'graphql-tag'
import { expect, it } from 'vitest'
import { request } from '../src/index.js'
import { setupTestServer } from './__helpers.js'

const ctx = setupTestServer()

it('accepts graphql DocumentNode as alternative to raw string', async () => {
  const mock = ctx.res({ body: { data: { foo: 1 } } })
  await request(
    ctx.url,
    gql`
      {
        query {
          users
        }
      }
    `
  )
  expect(mock).toMatchSnapshot()
})
