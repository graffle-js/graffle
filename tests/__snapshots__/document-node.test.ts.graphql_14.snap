// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts graphql DocumentNode as alternative to raw string 1`] = `
Object {
  "requests": Array [
    Object {
      "body": Object {
        "query": "{
  query {
    users
  }
}
",
      },
      "headers": Object {
        "accept": "*/*",
        "accept-encoding": "gzip,deflate",
        "connection": "close",
        "content-length": "45",
        "content-type": "application/json",
        "host": "DYNAMIC",
        "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
      },
      "method": "POST",
    },
  ],
  "spec": Object {
    "body": Object {
      "data": Object {
        "foo": 1,
      },
    },
  },
}
`;
